src/main/java/org/springframework/up/command/BootCommands.java

```diff
@@ -0,0 +1,205 @@
+		for (TemplateRepository templateRepository : templateRepositories) {
+			if (templateName.trim().equalsIgnoreCase(templateRepository.getName().trim())) {
+				// match - get url
+				String url = templateRepository.getUrl();
+				if (StringUtils.hasText(url)) {
+					return Optional.of(url);
+				}
+				break;
+			}
+		}
+		return Optional.empty();
+	}
+
+	private void generateFromUrl(String projectName, String url, String packageName) {
+
+		Path repositoryContentsPath = sourceRepositoryService.retrieveRepositoryContents(url);
+		if (!StringUtils.hasText(packageName))  {
+			packageName = this.upCliProperties.getDefaults().getPackageName();
+		}
+		if (StringUtils.hasText(packageName)) {
+			refactorPackage(packageName, repositoryContentsPath);
+		}
+		File projectDirectory = createProjectDirectory(projectName);
+		try {
+			FileSystemUtils.copyRecursively(repositoryContentsPath.toFile(), projectDirectory);
+		}
+		catch (IOException e) {
+			e.printStackTrace();
+			throw new UpException("Could not copy files.", e);
+		}
+
+		AttributedStringBuilder sb = new AttributedStringBuilder();
+		sb.append("Project " + projectName + " created in " + repositoryContentsPath)
+				.style(sb.style().foreground(AttributedStyle.GREEN));
+		System.out.println(sb.toAnsi());
+	}
+
+	private File createProjectDirectory(String projectName) {
+		File workingDirectory = IoUtils.getWorkingDirectory();
+		String projectNameToUse = projectName.replaceAll(" ", "_");
+		File projectDirectory = new File(workingDirectory, projectNameToUse);
+		IoUtils.createDirectory(projectDirectory);
+		logger.debug("Created directory " + projectDirectory);
+		return projectDirectory;
+	}
+
+	private void refactorPackage(String targetPackageName, Path workingPath) {
+		logger.debug("Refactoring to package name " + targetPackageName);
+		JavaParser javaParser = new Java11Parser.Builder().build();
+		FileTypeCollectingFileVisitor collector = new FileTypeCollectingFileVisitor(".java");
+		try {
+			Files.walkFileTree(workingPath, collector);
+		}
+		catch (IOException e) {
+			throw new UpException("Failed reading files in " + workingPath, e);
+		}
+		List<? extends SourceFile> compilationUnits = javaParser.parse(collector.getMatches(), null, null);
+		ResultsExecutor container = new ResultsExecutor();
+
+		//TODO derive fromPackage using location of existing @SpringBootApplication class, hardcode for now
+		String fromPackage = "com.example.up";
+		Recipe recipe = new ChangePackage(fromPackage, targetPackageName, true);
+		container.addAll(recipe.run(compilationUnits));
+		try {
+			container.execute();
+		}
+		catch (IOException e) {
+			throw new UpException("Error performing refactoring", e);
+		}
+
+		//TODO change groupId and artifactId
+	}
+
+	private void validate(String templateName, String url) {
+		if (StringUtils.hasText(templateName) && StringUtils.hasText(url)) {
+			System.err.println("templateName and url option can not be specified together");
```

Comment:
This is probably something we should add to spring-shell. Anyway, I'd just throw for now.